python/hooks/default.nix

```diff
@@ -55,33 +70,55 @@ let
           exit 0
         fi
 
-        ${toolDerivation}/bin/${toolName} ${configFlag} "$config_file" "$@" ${extraArgs}
+        ${execution}
       '';
       destination = "/bin/${toolName}";
     };
 
   blackWithConfig = toolDerivation: generateConfigurationRunner {
     inherit toolDerivation;
-    key = "black";
+    key = "tool.black";
     config = "black.toml";
     files = [
-      { path = "pyproject.toml"; key = "tool.black"; }
-      "setup.cfg"
-      { path = ./config/black.toml; key = "tool.black"; }
+      "pyproject.toml"
+      { path = "setup.cfg"; key = "black"; }
+      ./config/black.toml
+    ];
+  };
+
+  ruffWithConfig = toolDerivation: generateConfigurationRunner {
+    inherit toolDerivation;
+    # Ruff can only take the config argument for the sub commands
+    # check and format which is kind of annoying.
+    customExecution = ''
+      configArgs=""
+      if [[ " $* " =~ [[:blank:]]check[[:blank:]] ]] || [[ " $* " =~ [[:blank:]]format[[:blank:]] ]]; then
```

Comment:
Wonder if we should handle the fact that `check` is the default subcommand here. I.e. if you run `ruff .` now, it will not apply the config. Not sure how to do that though...